### ISSUE TITLE
Unability to reproduce adaptation results

### ISSUE THREAD

**Issue by JulesSanchez** [saltoricristiano/gipso-sfouda#1]
Hello,

I'm trying to reproduce the various results you presented in the paper. First, thanks for the availability of the code and the pre-trained model.
I manage to reproduce source results of synlidar2kitti, but the final results CSV is equal to 0 for all category, as if there was no adaptation. I used the synlidar2kitti_adaptation.yaml file changing only the path of the data. Is everything working properly for you. I computed the DIP features, and then run :
`CUBLAS_WORKSPACE_CONFIG=:4096:8 python adapt_online.py --config_file configs/adaptation/synlidar2kitti_adaptation.yaml --geometric_path experiments/dip_features/semantickitti/08`

Am I missing a step ? Is there any modification to run the yaml file ?

Best,

SANCHEZ Jules

**Comment by saltoricristiano** [comment]
Hi @JulesSanchez, 

thanks for your interest in our work!
Apparently, no, you are not missing any step...let's try to debug a bit.
Have you tried to run on the other directions? For example, have you been able to run Synth4D -> SemanticKITTI?

**Comment by JulesSanchez** [comment]
Hi @saltoricristiano 

I just tried running it, it yields the same results (difference = 0 in the final_main.csv). I joined the logs here, the pretrained model is well loaded as the source results are correct. I guess it has something to do with the adaptation step ?
![logs](https://user-images.githubusercontent.com/35608261/189676250-c5c200e9-6301-45ad-8999-0ada687a802a.png)

The only change I did in the code is to chance lines 283 and 284 in adaptation_online_single.py to cast matches to Long tensor as otherwise it was raising a type mismatch. Could it stem from here ? Sadly without this fix it raises the following error : 
![error](https://user-images.githubusercontent.com/35608261/189679619-c72e8223-86eb-4ec7-a461-9e5c7a8a666d.png)

Best and thanks for the help,

Jules

**Comment by saltoricristiano** [comment]
Hmmmm, so, I tested the code and it was working + other people wrote me saying that they have been able to reproduce the results.
Let's start from the beginning... ( sorry for the stupid questions but I want to be sure we are in teh same setup):

1) are you sure your python version is 3.8 and that you are using the same pytorch version (1.8.0)?
2) if not yet, did you try to run the code inside the docker container provided?

Cristiano

**Comment by saltoricristiano** [comment]
Hey @JulesSanchez,

I'm going to close the issue due to inactivity!
Feel free to comment it or open a new one in case you still need help!

**Comment by vijayM12** [comment]
> 

Hello @saltoricristiano I appreciate your work on the syn2real adaptation, Please help with the following issues
 I tried reproducing the adaptation result and I am facing the following issues (pytorch 1.9.0 with cuda 10.2 i am using)
1. I tried to Preprocess geometric features but getting the output as follows
![error_1](https://user-images.githubusercontent.com/93333400/212338611-131a4605-2b86-49f6-bb0d-3361aa1db413.png)
2. I tried to set wandb login I am facing issues with the login into my wandb account ( i have changed the wandb project name details in the config file as well)
![error_2](https://user-images.githubusercontent.com/93333400/212338935-45a88257-5001-4576-b671-51cbd31b4d22.png)
Please let me know if we can results without using wandb......

**Comment by saltoricristiano** [comment]
Hi @vijayM12,

I see you are running the code in colab.
From a high level, it looks like colab is not able to connect to the wandb server. However, I don't have experience with colab and I cannot helo in this direction.
You can try to change the logs of the project but it could be quite laborious. The easiest solution would be to put wandb to offline and check if it runs in offline mode.

**Comment by vijayM12** [comment]
> Hi @vijayM12,
> 
> I see you are running the code in colab. From a high level, it looks like colab is not able to connect to the wandb server. However, I don't have experience with colab and I cannot helo in this direction. You can try to change the logs of the project but it could be quite laborious. The easiest solution would be to put wandb to offline and check if it runs in offline mode.

Thanks for reply. Wandb offline is working. But facing issues due to limited storage in colab as each file in geometric scenes is 30mb and also in .npz format I tried to run the adaptation from synlidar2kitti. Could you please provide the wandb logs of your experiment classwise with miou and loss. so that I can compare after reproducing the results. Also any drive point access where we have the synlidar to kitti converted files for all sequences is present. I tried with only hundred samples due to storage issue. It would be helpful if you can provide above pcd files for all the sequnces in sylidar after prediction for my analysis

**Comment by saltoricristiano** [comment]
Hi @vijayM12 ,

wandb is used in the code only for logging the metrics during training. However, I strongly recommend you to use the evaluation protocol to evaluate the papeline as wandb sometimes may log slightly different averages. Moreover, in the project wandb is not used for evaluating the pipeline. You can find the evaluation results in the main paper.
For synlidar 2 kitti, you can find the dataset [here](https://github.com/xiaoaoran/SynLiDAR). We do not save point clouds converted in synlidar2kitti as we re-map the labels as you can see [here](https://github.com/saltoricristiano/gipso-sfouda/blob/main/utils/_resources/synlidar.yaml).
